src/v1/sources/shopify/webhookTransformations/serverSideUtlis.js

```diff
@@ -73,22 +85,32 @@ const getCartToken = (event) => event?.cart_token || null;
  */
 const setAnonymousId = async (message, event, metricMetadata) => {
   const anonymousId = getAnonymousIdFromAttributes(event);
+  const cartToken = getCartToken(event);
+  const cartTokenHash = cartToken ? uuidv5(cartToken) : null;
   if (isDefinedAndNotNull(anonymousId)) {
     message.anonymousId = anonymousId;
-  } else {
-    // if anonymousId is not present in note_attributes or note_attributes is not present, query redis for anonymousId
-    const cartToken = getCartToken(event);
     if (cartToken) {
```

Comment:
you can move condition on cartToken/cartTokenHash to addCartTokenHashToTraits function

Reply:
Also addCartTokenHashToTraits is default behaviour expected, why is this dependent on anonId ?